frontend/src/__tests__/cypress/cypress/e2e/projects/clusterStorage.cy.ts

```diff
@@ -0,0 +1,214 @@
+
+    cy.intercept(
+      {
+        method: 'PATCH',
+        pathname: '/api/k8s/apis/kubeflow.org/v1/namespaces/test-project/notebooks/test-notebook',
+      },
+      mockNotebookK8sResource({}),
+    ).as('addClusterStorage');
+
+    cy.intercept(
+      { pathname: '/api/k8s/api/v1/namespaces/test-project/persistentvolumeclaims' },
+      mockK8sResourceList([mockPVCK8sResource({})]),
+    );
+
+    addClusterStorageModal.findSubmitButton().click();
+    cy.wait('@addClusterStorage').then((interception) => {
+      expect(interception.request.url).to.include('?dryRun=All');
+      expect(interception.request.body).to.eql([
+        { op: 'add', path: '/spec/template/spec/volumes/-', value: { persistentVolumeClaim: {} } },
+        {
+          op: 'add',
+          path: '/spec/template/spec/containers/0/volumeMounts/-',
+          value: { mountPath: '/opt/app-root/src/data' },
+        },
+      ]);
+    });
+
+    cy.wait('@addClusterStorage').then((interception) => {
+      expect(interception.request.url).not.to.include('?dryRun=All');
+    });
+
+    cy.get('@addClusterStorage.all').then((interceptions) => {
+      expect(interceptions).to.have.length(2);
+    });
+  });
+
+  it('Renders the page with cluster storage table data', () => {
+    initInterceptors({});
+    clusterStorage.visit('test-project');
+    const clusterStorageRow = clusterStorage.getClusterStorageRow('Test Storage');
+    clusterStorageRow.shouldHaveStorageTypeValue('Persistent storage');
+    clusterStorageRow.findConnectedWorkbenches().should('have.text', 'No connections');
+    clusterStorageRow.toggleExpandByIndex(0);
+    clusterStorageRow.shouldHaveStorageSize('Max 5Gi');
+  });
+
+  it('Edit cluster storage', () => {
+    initInterceptors({});
+    clusterStorage.visit('test-project');
+    const clusterStorageRow = clusterStorage.getClusterStorageRow('Test Storage');
+    clusterStorageRow.findKebabAction('Edit storage').click();
+    updateClusterStorageModal.findNameInput().should('have.value', 'Test Storage');
+    updateClusterStorageModal.findPVSizeInput().should('have.value', '5');
+    updateClusterStorageModal.shouldHavePVSizeSelectValue('Gi');
+    updateClusterStorageModal.findPersistentStorageWarning().should('exist');
+    updateClusterStorageModal.findSubmitButton().should('be.enabled');
+    updateClusterStorageModal.findNameInput().fill('test-updated');
+
+    cy.intercept(
+      {
+        method: 'PUT',
+        pathname: '/api/k8s/api/v1/namespaces/test-project/persistentvolumeclaims/test-storage',
+      },
+      mockPVCK8sResource({}),
+    ).as('editClusterStorage');
+
+    updateClusterStorageModal.findSubmitButton().click();
+    cy.wait('@editClusterStorage').then((interception) => {
+      expect(interception.request.url).to.include('?dryRun=All');
+      expect(interception.request.body).to.eql({
+        kind: 'PersistentVolumeClaim',
+        apiVersion: 'v1',
+        metadata: {
```

Comment:
May be we could use `chai-subset` here after this [PR](https://github.com/opendatahub-io/odh-dashboard/pull/2643) gets merged.